cli/src/main/java/com/devonfw/tools/ide/url/model/folder/AbstractUrlFolder.java

```diff
@@ -61,6 +67,45 @@ public C getOrCreateChild(String name) {
     return this.childMap.computeIfAbsent(name, p -> newChild(name));
   }
 
+  @Override
+  public void deleteChild(String name) {
+
+    C child = this.childMap.remove(name);
+    if (child != null) {
+      delete(child.getPath());
+    }
+  }
+
+  private static void delete(Path path) {
+
+    LOG.debug("Deleting {}", path);
+    if (Files.exists(path)) {
+      try {
+        deleteRecursive(path);
+      } catch (IOException e) {
+        throw new RuntimeException("Failed to delete " + path);
+      }
+    } else {
+      LOG.warn("Could not delete file {} because it does not exist.", path);
+    }
+  }
```

Comment:
Move to FileAccess?

Reply:
Cannot until we have solved #404 since currently `FileAccessImpl` is based on `IdeContext` and uses that for logging.